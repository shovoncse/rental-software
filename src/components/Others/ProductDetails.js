import React from 'react';
import { List } from 'antd';
const ProductDetails = ({ selected }) => {
  return (
    <>
      <List size="small" bordered className="bg-sky-100">
        <List.Item key={1}>
          <span className="text-sm font-bold">{selected.name}</span>
        </List.Item>
        {selected.bookingsDays ? (
          <List.Item key={2}>
            <span className="font-bold">Bookd for:</span>{' '}
            {selected.bookingsDays} Days <br />
            <span className="font-bold">Period:</span>{' '}
            {selected.bookingsDates
              ? selected.bookingsDates[0] + ' to ' + selected.bookingsDates[1]
              : 'Not Found'}
            <br />
            <span className="font-bold">Mileage:</span> {selected.mileage || 0}
          </List.Item>
        ) : (
          <List.Item key={2}>
            <span className="font-semibold">Mileage:</span>{' '}
            {selected.mileage || '0'} <br />
            <span className="font-semibold">Price:</span> {selected.price}
            <br />
            <span className="font-semibold">Min Rent:</span> For{' '}
            <span className="font-bold">{selected.minimum_rent_period}</span>{' '}
            Day(s)
          </List.Item>
        )}
        <List.Item key={3}>
          {selected.needing_repair ? (
            <span className="text-sm text-red-500 font-bold">
              Need to Repair
            </span>
          ) : (
            <span className="text-sm font-bold text-green-700">
              Repairing Not Necessary
            </span>
          )}
        </List.Item>
      </List>
    </>
  );
};

export default ProductDetails;
