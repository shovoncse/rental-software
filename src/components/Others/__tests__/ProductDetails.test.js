import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import ProductDetails from '../ProductDetails';

afterEach(cleanup);

describe('<ProductDetails />', () => {
  const selected = { name: '', mileage: '', minimum_rent_period: '' };
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ProductDetails selected={selected}></ProductDetails>, div);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<ProductDetails selected={selected}></ProductDetails>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
