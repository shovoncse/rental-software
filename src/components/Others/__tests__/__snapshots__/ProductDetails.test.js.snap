// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetails /> matches snapshot 1`] = `
<div
  className="ant-list ant-list-sm ant-list-split ant-list-bordered bg-sky-100"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <li
        className="ant-list-item"
      >
        <span
          className="text-sm font-bold"
        >
          Not Found
        </span>
      </li>
      <li
        className="ant-list-item ant-list-item-no-flex"
      >
        <span
          className="font-semibold"
        >
          Mileage:
        </span>
         
        0
         
        <br />
        <span
          className="font-semibold"
        >
          Price:
        </span>
         
        <br />
        <span
          className="font-semibold"
        >
          Min Rent:
        </span>
         For
         
        <span
          className="font-bold"
        >
          
        </span>
         
        Day(s)
      </li>
      <li
        className="ant-list-item"
      >
        <span
          className="text-sm font-bold text-green-700"
        >
          Repairing Not Necessary
        </span>
      </li>
    </div>
  </div>
</div>
`;
